Add "resource identifier" and "percent encoding"

aphillips opened the pull request:
This is part of I18N-ACTION-1212. Copied these two definitions from specdev. Note that these remain definitions there as well.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/i18n-glossary/pull/18.html" title="Last updated on Nov 3, 2022, 6:05 PM UTC (9dea8ce)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/i18n-glossary/18/0a3f668...aphillips:9dea8ce.html" title="Last updated on Nov 3, 2022, 6:05 PM UTC (9dea8ce)">Diff</a>

aphillips committed:
Add "resource identifier" and "percent encoding"

This is part of I18N-ACTION-1212. Copied these two definitions from specdev. Note that these remain definitions there as well.

xfq commented on index.html:
```diff
@@ -331,12 +331,14 @@ <h2>Glossary</h2>
-<p><dfn data-lt="resource|resources" id="def_resource" class="lint-ignore export">Resource</dfn>. In the context of W3C Internationalization documents,   a given document, file, or protocol "message" which includes both the <a href="#def_localizable_text" class="termref">localizable text</a> as well as the <a href="#def_syntactic_content" class="termref">syntactic content</a> such as identifiers  ...
+<p><dfn data-lt="resource|resources" id="def_resource" class="lint-ignore export">Resource</dfn>. In the context of W3C Internationalization documents,   a given document, file, or protocol "message" which includes both the <a href="#def_localizable_text" class="termref">localizable text</a> as well as the <a href="#def_syntactic_content" class="termref">syntactic content</a> such as identifiers  ...
+
+<p><dfn id="dfn_resid" data-lt="resource identifier|resource identifiers" class="lint-ignore export">Resource identifier</dfn>. A resource identifier is a compact string of characters for identifying an abstract or physical <a>resource</a>. On the Web, this mostly means various types of Universal Resource Identifiers (or <em>URIs</em>). For wire formats, [[RFC3986]] defines the structure and seri ...
```
We should add a `class="termref"` to "resource" to make the styling more consistent.

r12a replied:
Internationalizationed is an interesting new word, but i don't think it's actually in the title of that document.  And the verb 'describe' should have an 's' at the end, since the subject is a document.

aphillips replied:
This is the actual title of RFC3987: the word *internationalized* ***is*** the first word :-) https://www.rfc-editor.org/rfc/rfc3987

aphillips replied:
added termref

r12a replied:
Indeed, the first word is 'Internationalized', not 'Internationalizationed'. Still needs fixing.

r12a commented on index.html:
```diff
@@ -331,12 +331,14 @@ <h2>Glossary</h2>
 <p><dfn class="lint-ignore export">Paragraph direction</dfn>. The initial <a>base direction</a>  of  a paragraph or string, which resolves to either <em>left-to-right</em> or <em>right-to-left</em>. Nested embedding controls may be used to change the direction of an inline range of text, but the paragraph direction sets the starting point which the <a>Unicode Bidirectional Algorithm</a> uses  to  ...
 
-
+<p><dfn class="lint-ignore export" id="dfn_percent-encoding" data-lt="percent-encoding|percent encoding|%HH encoding|percent encoded">Percent-encoding</dfn>. Percent encoding is the escaping mechanism defined by URI [[RFC3986]] for the encoding of arbitrary byte values not otherwise permitted into a URI. For example, if a user wishes to include the character <span class="codepoint" translate="no" ...
```
fwiw 'include into' sounds a bit wierd to me – in my dialect of English i'd say 'include in'

aphillips replied:
fixed.

aphillips committed:
Fix location of one item; add abjad

Moved `Unicode code point` to the letter `U` where it belongs.

Added the Unicode definition of abjad.

aphillips committed:
change include into -> include in

aphillips committed:
Fix typo 'describes'

aphillips commented:
This needs to be merged before I can make additional changes to specdev. @r12a can you see if you're satisfied by these changes?

r12a commented:
We may need to improve the definition of abjad, since actually long vowels are represented in most abjads. Otherwise, once you fix Internationalizationed i think we can merge.

r12a commented:
Don't change abjad for now.  I'll update my own definition and suggest changes.

aphillips committed:
Add 'surrogate' and update 'supplementary character'

Defined 'surrogate' and its friend 'surrogate pair'. That definition is taking from the Unicode glossary with light editing.

Added `supplementary code point' to `supplementary characters`. Cleaned up the definition lightly, since we can now refer to `surrogate`.

aphillips committed:
Doh. The I18nized typo.

aphillips commented:
@r12a thanks: I missed that the word was horked in that way. 

Note that the definition for 'abjad' is verbatim from the Unicode glossary. I'll merge this and we can fix (both) later.

aphillips committed:
pairs -> pair
